viam_cartographer_test.go

```diff
@@ -142,110 +158,35 @@ func TestNew(t *testing.T) {
-		timestamp2, err := svc.GetLatestMapInfo(context.Background())
+		_, componentReference, err := svc.GetPosition(context.Background())
 		test.That(t, err, test.ShouldBeNil)
-		test.That(t, timestamp1.After(_zeroTime), test.ShouldBeTrue)
-		test.That(t, timestamp1, test.ShouldResemble, timestamp2)
-
-		test.That(t, svc.Close(context.Background()), test.ShouldBeNil)
-	})
-
-	t.Run("Successful creation of cartographer slam service in non localization mode", func(t *testing.T) {
-		termFunc := testhelper.InitTestCL(t, logger)
-		defer termFunc()
+		test.That(t, componentReference, test.ShouldEqual, "replay_sensor")
 
-		dataDirectory, fsCleanupFunc := testhelper.InitInternalState(t)
-		defer fsCleanupFunc()
+		pcmFunc, err := svc.GetPointCloudMap(context.Background())
+		test.That(t, err, test.ShouldBeNil)
 
-		attrCfg := &vcConfig.Config{
-			Sensors:       []string{"good_lidar"},
-			ConfigParams:  map[string]string{"mode": "2d"},
-			DataDirectory: dataDirectory,
-			MapRateSec:    &testMapRateSec,
-		}
+		pcm, err := slam.HelperConcatenateChunksToFull(pcmFunc)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, pcm, test.ShouldNotBeNil)
 
-		svc, err := testhelper.CreateSLAMService(t, attrCfg, logger)
+		isFunc, err := svc.GetInternalState(context.Background())
 		test.That(t, err, test.ShouldBeNil)
 
-		cs, ok := svc.(*viamcartographer.CartographerService)
-		test.That(t, ok, test.ShouldBeTrue)
-		test.That(t, cs.SlamMode, test.ShouldEqual, cartofacade.UpdatingMode)
+		is, err := slam.HelperConcatenateChunksToFull(isFunc)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, is, test.ShouldNotBeNil)
 
 		timestamp1, err := svc.GetLatestMapInfo(context.Background())
 		test.That(t, err, test.ShouldBeNil)
-		_, err = svc.GetPointCloudMap(context.Background())
-		test.That(t, err, test.ShouldBeNil)
 		timestamp2, err := svc.GetLatestMapInfo(context.Background())
 		test.That(t, err, test.ShouldBeNil)
-
 		test.That(t, timestamp1.After(_zeroTime), test.ShouldBeTrue)
-		test.That(t, timestamp2.After(timestamp1), test.ShouldBeTrue)
+		test.That(t, timestamp1, test.ShouldResemble, timestamp2)
 
 		test.That(t, svc.Close(context.Background()), test.ShouldBeNil)
 	})
 
-	t.Run("Fails to create cartographer slam service with no sensor when feature flag enabled", func(t *testing.T) {
-		termFunc := testhelper.InitTestCL(t, logger)
-		defer termFunc()
-
-		dataDirectory, fsCleanupFunc := testhelper.InitInternalState(t)
-		defer fsCleanupFunc()
-
-		attrCfg := &vcConfig.Config{
-			Sensors:       []string{},
-			ConfigParams:  map[string]string{"mode": "2d"},
-			DataDirectory: dataDirectory,
-		}
-
-		svc, err := testhelper.CreateSLAMService(t, attrCfg, logger)
-		test.That(t, err, test.ShouldBeError, errors.New("error validating \"path\": \"sensors\" must not be empty"))
-		test.That(t, svc, test.ShouldBeNil)
-	})
-
-	t.Run("Failed creation of cartographer slam service with more than one sensor when feature flag enabled", func(t *testing.T) {
```

Comment:
removed as it is already covered by `Failed creation of cartographer slam service with more than one sensor`